From 0822063abf318cb593ed1fcc6d41525a045392d5 Mon Sep 17 00:00:00 2001
From: oblique <psyberbits@gmail.com>
Date: Mon, 25 Feb 2013 14:01:35 +0200
Subject: [PATCH] change i3locks colors to my taste

---
 i3lock.c           |  2 +-
 unlock_indicator.c | 29 +++++++++--------------------
 2 files changed, 10 insertions(+), 21 deletions(-)

diff --git a/i3lock.c b/i3lock.c
index df07d90..cd37443 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -36,7 +36,7 @@
 
 /* We need this for libxkbfile */
 static Display *display;
-char color[7] = "ffffff";
+char color[7] = "101010";
 uint32_t last_resolution[2];
 xcb_window_t win;
 static xcb_cursor_t cursor;
diff --git a/unlock_indicator.c b/unlock_indicator.c
index ab4d56e..730e86d 100644
--- a/unlock_indicator.c
+++ b/unlock_indicator.c
@@ -121,30 +121,18 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
                   0 /* start */,
                   2 * M_PI /* end */);
 
-        /* Use the appropriate color for the different PAM states
-         * (currently verifying, wrong password, or default) */
-        switch (pam_state) {
-            case STATE_PAM_VERIFY:
-                cairo_set_source_rgba(ctx, 0, 114.0/255, 255.0/255, 0.75);
-                break;
-            case STATE_PAM_WRONG:
-                cairo_set_source_rgba(ctx, 250.0/255, 0, 0, 0.75);
-                break;
-            default:
-                cairo_set_source_rgba(ctx, 0, 0, 0, 0.75);
-                break;
-        }
+        cairo_set_source_rgba(ctx, 0x10/255.0, 0x10/255.0, 0x10/255.0, 0.75);
         cairo_fill_preserve(ctx);
 
         switch (pam_state) {
             case STATE_PAM_VERIFY:
-                cairo_set_source_rgb(ctx, 51.0/255, 0, 250.0/255);
+                cairo_set_source_rgb(ctx, 0x5f/255.0, 0x87/255.0, 0xd7/255.0);
                 break;
             case STATE_PAM_WRONG:
-                cairo_set_source_rgb(ctx, 125.0/255, 51.0/255, 0);
+                cairo_set_source_rgb(ctx, 0xcb/255.0, 0x00/255.0, 0x52/255.0);
                 break;
             case STATE_PAM_IDLE:
-                cairo_set_source_rgb(ctx, 51.0/255, 125.0/255, 0);
+                cairo_set_source_rgb(ctx, 0x66/255.0, 0xff/255.0, 0x66/255.0);
                 break;
         }
         cairo_stroke(ctx);
@@ -162,14 +150,16 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
 
         cairo_set_line_width(ctx, 10.0);
 
-        /* Display a (centered) text of the current PAM state. */
+        /* Display a (centered and colored) text of the current PAM state. */
         char *text = NULL;
         switch (pam_state) {
             case STATE_PAM_VERIFY:
                 text = "verifying…";
+                cairo_set_source_rgb(ctx, 0x5f/255.0, 0x87/255.0, 0xd7/255.0);
                 break;
             case STATE_PAM_WRONG:
                 text = "wrong!";
+                cairo_set_source_rgb(ctx, 0xcb/255.0, 0x00/255.0, 0x52/255.0);
                 break;
             default:
                 break;
@@ -179,7 +169,6 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
             cairo_text_extents_t extents;
             double x, y;
 
-            cairo_set_source_rgb(ctx, 0, 0, 0);
             cairo_set_font_size(ctx, 28.0);
 
             cairo_text_extents(ctx, text, &extents);
@@ -206,10 +195,10 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
                       highlight_start + (M_PI / 3.0));
             if (unlock_state == STATE_KEY_ACTIVE) {
                 /* For normal keys, we use a lighter green. */
-                cairo_set_source_rgb(ctx, 51.0/255, 219.0/255, 0);
+                cairo_set_source_rgb(ctx, 0x5f/255.0, 0x87/255.0, 0xd7/255.0);
             } else {
                 /* For backspace, we use red. */
-                cairo_set_source_rgb(ctx, 219.0/255, 51.0/255, 0);
+                cairo_set_source_rgb(ctx, 0xcb/255.0, 0x00/255.0, 0x52/255.0);
             }
             cairo_stroke(ctx);
 
-- 
1.8.1.4

