From 03b49c3ddeaa8d8df963004d38a65479b9676e9b Mon Sep 17 00:00:00 2001
From: oblique <psyberbits@gmail.com>
Date: Fri, 13 Mar 2015 22:56:46 +0200
Subject: [PATCH] change i3locks colors to my taste

---
 i3lock.c           |  2 +-
 unlock_indicator.c | 29 +++++++++--------------------
 2 files changed, 10 insertions(+), 21 deletions(-)

diff --git a/i3lock.c b/i3lock.c
index 7f8218e..8c93222 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -46,7 +46,7 @@
 typedef void (*ev_callback_t)(EV_P_ ev_timer *w, int revents);
 
 /* We need this for libxkbfile */
-char color[7] = "ffffff";
+char color[7] = "101010";
 int inactivity_timeout = 30;
 uint32_t last_resolution[2];
 xcb_window_t win;
diff --git a/unlock_indicator.c b/unlock_indicator.c
index e543f96..1e15257 100644
--- a/unlock_indicator.c
+++ b/unlock_indicator.c
@@ -147,30 +147,18 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
                   0 /* start */,
                   2 * M_PI /* end */);
 
-        /* Use the appropriate color for the different PAM states
-         * (currently verifying, wrong password, or default) */
-        switch (pam_state) {
-            case STATE_PAM_VERIFY:
-                cairo_set_source_rgba(ctx, 0, 114.0/255, 255.0/255, 0.75);
-                break;
-            case STATE_PAM_WRONG:
-                cairo_set_source_rgba(ctx, 250.0/255, 0, 0, 0.75);
-                break;
-            default:
-                cairo_set_source_rgba(ctx, 0, 0, 0, 0.75);
-                break;
-        }
+        cairo_set_source_rgba(ctx, 0x10/255.0, 0x10/255.0, 0x10/255.0, 0.75);
         cairo_fill_preserve(ctx);
 
         switch (pam_state) {
             case STATE_PAM_VERIFY:
-                cairo_set_source_rgb(ctx, 51.0/255, 0, 250.0/255);
+                cairo_set_source_rgb(ctx, 0x5f/255.0, 0x87/255.0, 0xd7/255.0);
                 break;
             case STATE_PAM_WRONG:
-                cairo_set_source_rgb(ctx, 125.0/255, 51.0/255, 0);
+                cairo_set_source_rgb(ctx, 0xcb/255.0, 0x00/255.0, 0x52/255.0);
                 break;
             case STATE_PAM_IDLE:
-                cairo_set_source_rgb(ctx, 51.0/255, 125.0/255, 0);
+                cairo_set_source_rgb(ctx, 0x66/255.0, 0xff/255.0, 0x66/255.0);
                 break;
         }
         cairo_stroke(ctx);
@@ -193,14 +181,15 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
         /* We don't want to show more than a 3-digit number. */
         char buf[4];
 
-        cairo_set_source_rgb(ctx, 0, 0, 0);
         cairo_set_font_size(ctx, 28.0);
         switch (pam_state) {
             case STATE_PAM_VERIFY:
                 text = "verifying…";
+                cairo_set_source_rgb(ctx, 0x5f/255.0, 0x87/255.0, 0xd7/255.0);
                 break;
             case STATE_PAM_WRONG:
                 text = "wrong!";
+                cairo_set_source_rgb(ctx, 0xcb/255.0, 0x00/255.0, 0x52/255.0);
                 break;
             default:
                 if (show_failed_attempts && failed_attempts > 0){
@@ -210,7 +199,7 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
                         snprintf(buf, sizeof(buf), "%d", failed_attempts);
                         text = buf;
                     }
-                    cairo_set_source_rgb(ctx, 1, 0, 0);
+                    cairo_set_source_rgb(ctx, 0xcb/255.0, 0x00/255.0, 0x52/255.0);
                     cairo_set_font_size(ctx, 32.0);
                 }
                 break;
@@ -244,10 +233,10 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
                       highlight_start + (M_PI / 3.0));
             if (unlock_state == STATE_KEY_ACTIVE) {
                 /* For normal keys, we use a lighter green. */
-                cairo_set_source_rgb(ctx, 51.0/255, 219.0/255, 0);
+                cairo_set_source_rgb(ctx, 0x5f/255.0, 0x87/255.0, 0xd7/255.0);
             } else {
                 /* For backspace, we use red. */
-                cairo_set_source_rgb(ctx, 219.0/255, 51.0/255, 0);
+                cairo_set_source_rgb(ctx, 0xcb/255.0, 0x00/255.0, 0x52/255.0);
             }
             cairo_stroke(ctx);
 
-- 
2.3.0

